core/modules/block/block.module

```diff
@@ -89,7 +89,7 @@ function block_block_info() {
   foreach ($config_names as $config_name) {
     $block = config_get($config_name);
     $blocks[$block['delta']] = $block;
-    $blocks[$block['delta']]['description'] = t('A reusable custom block.');
+    $blocks[$block['delta']]['description'] = strlen($block['description']) ? $block['description'] : '';
```

Comment:
...also might be better to make that check be `strlen($block['description']) != 0`